tests/output.rs

```diff
@@ -227,3 +246,94 @@ fn test_specify_multiple_files() {
         "#,
     );
 }
+
+#[test]
+fn test_invalid_query_inline() {
+    assert_failure_output(
+        "rust_project",
+        r#"
+            $ tree-sitter-grep --query-source '(function_itemz) @function_item' --language rust
+            error: invalid query
+        "#,
+    );
+}
+
+#[test]
+fn test_invalid_query_file() {
+    assert_failure_output(
+        "rust_project",
+        r#"
+            $ tree-sitter-grep --query-file ./function-itemz.scm --language rust
+            error: invalid query
+        "#,
+    );
+}
+
+#[test]
+fn test_no_query_specified() {
+    assert_failure_output(
+        "rust_project",
+        r#"
+            $ tree-sitter-grep --language rust
+            error: the following required arguments were not provided:
+              <--query-file <PATH_TO_QUERY_FILE>|--query-source <QUERY_SOURCE>>
+            
+            Usage: tree-sitter-grep --language <LANGUAGE> <--query-file <PATH_TO_QUERY_FILE>|--query-source <QUERY_SOURCE>> [PATHS]...
+            
+            For more information, try '--help'.
+        "#,
+    );
+}
+
+#[test]
+fn test_invalid_capture_name() {
+    assert_failure_output(
+        "rust_project",
+        r#"
+            $ tree-sitter-grep --query-source '(function_item) @function_item' --language rust --capture function_itemz
+            error: invalid capture name 'function_itemz'
+        "#,
+    );
+}
+
+#[test]
+fn test_auto_language_single_known_language_encountered() {
+    assert_sorted_output(
+        "rust_project",
+        r#"
+            $ tree-sitter-grep --query-source '(function_item) @function_item'
+            src/helpers.rs:1:pub fn helper() {}
+            src/lib.rs:3:pub fn add(left: usize, right: usize) -> usize {
+            src/lib.rs:4:    left + right
+            src/lib.rs:5:}
+            src/lib.rs:12:    fn it_works() {
+            src/lib.rs:13:        let result = add(2, 2);
+            src/lib.rs:14:        assert_eq!(result, 4);
+            src/lib.rs:15:    }
+            src/stop.rs:1:fn stop_it() {}
+        "#,
+    );
+}
+
+#[test]
+fn test_auto_language_multiple_parseable_languages() {
+    assert_sorted_output(
+        "mixed_project",
+        r#"
+            $ tree-sitter-grep --query-source '(arrow_function) @arrow_function'
+            javascript_src/index.js:1:const js_foo = () => {}
+            typescript_src/index.tsx:1:const foo = () => {}
```

Comment:
Pretty cool cross-language queries!

(no idea how often that would be practically useful but sort of technically interesting)